### PR TITLE
Fix pod delete perf

### PR DESCRIPTION
**Pull request by trozet**
- Enables scale metrics correctly in KIND with single zone per node
- Enables pprof when enabling metrics with KIND
- Prefers to use IPs provided in pod status IPs, and falls back to annotation when needed
- For searching for colliding pods, restrict the search to only pods on the node where the original pod lived, as long as the topology is layer 3, and the pods are not affected by live migration.

### CODE REVIEW

**Comment by trozet** [comment]
adding hold until we have downstream perf/scale results. Some preliminary results on my local testing are promising, but lets wait for complete results before we merge. Ready for review though @jcaamano @tssurya

File: go-controller/pkg/ovn/base_network_controller_pods.go
Comment (jcaamano) [inline go-controller/pkg/ovn/base_network_controller_pods.go:271]: nit: typo

File: go-controller/pkg/ovn/base_network_controller_pods.go
Comment (jcaamano) [inline go-controller/pkg/ovn/base_network_controller_pods.go:290]: For the default network the topology is an empty string but you still want it covered in this check right?

File: go-controller/pkg/ovn/base_network_controller_pods.go
Comment (jcaamano) [inline go-controller/pkg/ovn/base_network_controller_pods.go:289]: Hmm not sure about this

This could all be subnet based right? We would just care about pods in the same subnet. And I wouldn't be sure if actually checking for subnet containment could be faster than comparing strings, the former could be less comparisons.

Anyway, I am fine with this as is an improvement in any case.

File: go-controller/pkg/ovn/base_network_controller_pods.go
Comment (trozet) [reply go-controller/pkg/ovn/base_network_controller_pods.go:290]: wait why do we set topology type as empty for default network...shouldnt we set it to L3?

File: go-controller/pkg/ovn/base_network_controller_pods.go
Comment (trozet) [reply go-controller/pkg/ovn/base_network_controller_pods.go:289]: from what I saw we would have to call  virtualMachineReady to know if the machine has migrated or not. It does a search based on pod selector, so maybe its not that bad. Either way I didnt really care that much because the optimizing for the live migration case is so small its not really worth it.

File: go-controller/pkg/ovn/base_network_controller_pods.go
Comment (jcaamano) [reply go-controller/pkg/ovn/base_network_controller_pods.go:290]: I guess because it was conceptually thought as a different topology given that it has all the extra things that the secondary network doesn't have. And because the NetConf of the default network doesn't really have the attribute `TopologyType` so the natural default value for that is the the zero value for string.

This is an example of the typical check we have been using for this same thing
```
func (na *NodeAllocator) hasNodeSubnetAllocation() bool {
	// we only allocate subnets for L3 secondary network or default network
	return na.netInfo.TopologyType() == types.Layer3Topology || !na.netInfo.IsSecondary()
}
```

File: go-controller/pkg/util/multi_network.go
Comment (trozet) [reply go-controller/pkg/util/multi_network.go:101]: I did look at usage in the code. I was mainly looking at where using TopologyType would break primary network if the value wasnt empty. Most of the usage for TopologyType is guarded by IsSecondary. I think it is safe to make this change now. I can clean up your example if I push another change, or if you require it to merge.

File: go-controller/pkg/ovn/base_network_controller_pods.go
Comment (trozet) [reply go-controller/pkg/ovn/base_network_controller_pods.go:291]: ok as we spoke about on slack, I'd prefer to make minimal changes at this point (yeah I'm paranoid). I added a comment with a TODO to fix this later.

File: go-controller/pkg/ovn/base_network_controller_pods.go
Comment (trozet) [reply go-controller/pkg/ovn/base_network_controller_pods.go]: I updated with a comment to mention the specific scenario im chasing after here, where the pod status IPs are not yet available so we want to avoid an unnecessary unmarshal if the pod is not in the same subnet as the one we are looking for.

**Review by jcaamano — APPROVED** [review]

**Pull request merged**
